### DISCUSSION TITLE
Hide some buttons in the dropdown menu

### DISCUSSION THREAD

**Discussion by ivictor777** [Laravel-Backpack/community-forum#973] [Ideas]
Hello. Can you make it possible in the table to hide not all buttons in the dropdown list, but only some of them?

**Comment by karandatwani92** [comment]
Hey @ivictor777 
You can handle access to operations per entry basis using `setAccessCondition()` closure. 
```php
$this->crud->operation('list', function() {
        $this->crud->setAccessCondition(['update', 'delete'], function ($entry) {
            return $entry->id===1 ? true : false;
    });
});
```
https://backpackforlaravel.com/docs/6.x/crud-operations#handling-access-to-operations

**Reply by karandatwani92** [reply]
To deny access to all entries, simply use the following:
```php
$this->crud->denyAccess('operation_name');
$this->crud->denyAccess(['update', 'create', 'delete']);
```

**Comment by ivictor777** [comment]
Hi @karandatwani92
That's not what I meant. 
In Backpack, if I have a lot of buttons, I can hide them in a drop-down menu as shown in the screenshot:

![](https://user-images.githubusercontent.com/33960976/228809544-0d5a0d94-9195-4f45-9e20-e9ea32932f49.png)

I can do this:
`b) For a specific CrudController, in its setupListOperation() define CRUD::setOperationSetting('lineButtonsAsDropdown', true);`

and all my buttons will be hidden.

However, I cannot move only part of the buttons to the drop-down list. This would be convenient in my project, since there are a lot of buttons, and some are used often, others rarely. It would be great to have a More (...) button that hides rarely used buttons.

**Reply by karandatwani92** [reply]
Hey @pxpm 

It seems to be a Feature Request. Do we have an API to move a few buttons to the list and keep remaining inline?

**Reply by pxpm** [reply]
Indeed, this would be a Feature Request. 
Feel free to add it to the Feature List, and if more people vote on it, we may be able to think about implementing it. 

Since this bit of code is also part of Backpack/CRUD (MIT and open-source), if @ivictor777 wants to contribute feel free to send a PR and I will review it. (without time promises). 

If the PR is clean and it does not add more maintenance burden I think it's a reasonable thing that could help more panels. 
If it gets the logic too complicated we may NOT do it. I've never looked at the problem so I am not sure how what would be needed to accommodate such change. Probably `datatables_logic.blade.php` is the place where this would live. 

At the moment we can't prioritize working on this. Sorry. 🙏

**Comment by karandatwani92** [comment]
Closing in favor of https://github.com/Laravel-Backpack/community-forum/discussions/751#discussioncomment-9313913
